### PR TITLE
Feature/as a user i should be able to sign up to create an account

### PR DESCRIPTION
**Pull request by frankvielma**
The PR adds a sign up form for creating an account based on the criteria in this  [card](https://trello.com/c/UKK6lTu9/11-as-a-user-i-should-be-able-to-sign-up-to-create-an-account).

**Description**
I need to provide an email & password that would create an account into the system. No need to confirm the account after this step.

**Acceptance Criteria**

- [x] Given that I am logged out
- [x] When I go to the homepage
- [x] Then I should land on the login page and see a sign up link
- [x] When I click it
---------------------------------------------------
- [x] Then I should be redirected to the sign up page and see a form to sign up
- [x] When I fill the form correctly
- [x] Then my account should be created and I should be redirected to the logged in homepage
- [x] When I fill the form incorrectly
- [x] Then I should see an error message indicating my mistakes
---------------------------------------------------
- [x] Then I should also see a sign in link
- [x] When I click it
- [x] Then I should be redirected to the sign in page

[video_signup.webm](https://github.com/frankvielma/rs-blackmarket-fv/assets/412081/02efb608-4bc8-47e0-af20-07d4e1f24707)

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <div className="mt-16">
+        <Image
+          src="/images/rs_logo.png"
+          height={162}
+          width={162}
+          alt="Rs Logo"
+        />
+      </div>
+      <div className="mt-20 text-5xl font-semibold sm:text-[6.25rem]">
+        Black Market
```
Comment (dariokozicki) [inline app/page.tsx:19]: Consider adding i18n to the project

File: app/signup/page.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          <Button
+            variant="primary"
+            option="disabled"
+          >
+            Sign up
+          </Button>
+        </div>
+
+        <div className="mt-5 text-center">
+          By signing up, you accept the{' '}
```
Comment (dariokozicki) [inline app/signup/page.tsx:45]: With i18n you can avoid the need for {' '}

File: components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  variant: 'primary' | 'outline'
+  option: 'default' | 'disabled'
+}
+
+export default function Button({ variant, option, ...props }: Props) {
```
Comment (dariokozicki) [inline components/Button.tsx:6]: What do you think about using forwardRef to declare it?

File: components/InputField.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+export default function InputField({
+  type = '',
+  id = '',
+  label = '',
+  placeholder = '',
+}) {
+  return (
```
Comment (dariokozicki) [inline components/InputField.tsx:7]: What do you think about using forwardRef to declare it?
That way when you implement the signup/login forms you can straight up the {...register('email')} functionality that react-hook-form provides

File: components/InputPassword.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import Image from 'next/image'
+import { useState } from 'react'
+
+export default function InputPassword({
```
Comment (dariokozicki) [inline components/InputPassword.tsx:4]: Same with forwardRef

File: styles/globals.css [hunk 6]
```diff
@@ -26,5 +18,59 @@
 
 body {
   background-color: black;
+  color: white;
   font-size: 1.5em;
 }
+
+.custom-input {
+  @apply h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray;
```
Comment (dariokozicki) [inline styles/globals.css:26]: Warning: @apply should be used with caution https://twitter.com/adamwathan/status/1559250403547652097?lang=en

**Review by dariokozicki — COMMENTED** [review]
Great work! A few suggestions

File: app/page.tsx (hunk 1 above)
Comment (frankvielma) [reply app/page.tsx:19]: :+1:

File: app/signup/page.tsx (hunk 2 above)
Comment (frankvielma) [reply app/signup/page.tsx:45]: :+1:

File: components/Button.tsx (hunk 3 above)
Comment (frankvielma) [reply components/Button.tsx:6]: I will do it !

File: components/InputField.tsx (hunk 4 above)
Comment (frankvielma) [reply components/InputField.tsx:7]: :+1:

File: components/InputPassword.tsx (hunk 5 above)
Comment (frankvielma) [reply components/InputPassword.tsx:4]: :+1:

File: styles/globals.css (hunk 6 above)
Comment (frankvielma) [reply styles/globals.css:26]: :+1:

**Review by dariokozicki — APPROVED** [review]

**Pull request merged**